test/rules/unnecessary_escapes_in_string.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N unnecessary_escapes_in_string`
+
+f(o){
```

Comment:
It would be good to add at least one example of a raw string and at least one example using a multi-line string.

Reply:
I don't know how to mark a `// LINT` with multiline string :-(

Reply:
done for raw string

Reply:
Multiline strings aren't required to contain end of line markers, so you could do something like
```
f('''_\'_'''); // LINT
f('''_\''''); // OK
f('''\'_'''); // LINT
f('''_\"_'''); // LINT

f("""_\"_"""); // LINT
f("""_\""""); // OK
f("""\"_"""); // LINT
f("""_\'_"""); // LINT
```

Reply:
Sure my bad :)

Reply:
Actually multiline strings are a little harder to handle.

```dart
f('''_\'_'''); // LINT
f('''_'\''_'''); // OK or it cause error
```

Reply:
For now `'''_\'_'''` don't trigger a lint. Only `'''_\"_'''` triggers a lint.

Reply:
```
f('''_'\''_'''); // OK or it cause error
```
That would need to be OK because removing the backslash would result in parse errors.

> For now `'''_\'_'''` don't trigger a lint.

I think that's reasonable, given the complexity of multi-line strings.

Reply:
I think it would still be good to add the test cases with comments explaining the desired outcome in cases where the rule doesn't currently produce that outcome. If nothing else we should add the cases that are handled correctly.

Reply:
Finally I made the changes for multiline strings.

Reply:
Thanks! I really was only concerned with not forgetting the corner cases, but it's great to have them handled.